tests/src/core/testqgsprovidermetadata.cpp

```diff
@@ -0,0 +1,128 @@
+/***************************************************************************
+                         testqgsprovidermetadata.cpp
+                         ---------------
+    begin                : April 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgstest.h"
+#include <QObject>
+#include <qgsprovidermetadata.h>
+#include <qgsproviderregistry.h>
+
+class TestQgsProviderMetadata: public QObject
+{
+    Q_OBJECT
+
+  public:
+    TestQgsProviderMetadata() = default;
+
+  private slots:
+    void initTestCase();// will be called before the first testfunction is executed.
+    void cleanupTestCase();// will be called after the last testfunction was executed.
+    void init(); // will be called before each testfunction is executed.
+    void cleanup(); // will be called after every testfunction.
+
+    void checkBoolParameterSetting();
+
+  private:
+    QgsProviderMetadata *mMetadata = nullptr;
+};
+
+void TestQgsProviderMetadata::initTestCase()
+{
+  //
+  // Runs once before any tests are run
+  //
+  // init QGIS's paths - true means that all path will be inited from prefix
+  QgsApplication::init();
+  QgsApplication::initQgis();
+  QgsApplication::showSettings();
+
+}
+
+void TestQgsProviderMetadata::init()
+{
+  //create some objects that will be used in all tests...
+  //create a temporal object that will be used in all tests...
+
+  mMetadata = QgsProviderRegistry::instance()->providerMetadata( QStringLiteral( "raster" ) );
+}
+
+void TestQgsProviderMetadata::cleanup()
+{
+}
+
+void TestQgsProviderMetadata::cleanupTestCase()
+{
+  QgsApplication::exitQgis();
+}
+
+void TestQgsProviderMetadata::checkBoolParameterSetting()
```

Comment:
Perfect! 
If you want, you can also use https://doc.qt.io/qt-5/qttestlib-tutorial2-example.html

Reply:
ooh that is so cool!